src/psyclone/domain/gocean/kernel/psyir.py

```diff
@@ -0,0 +1,814 @@
+            '''
+            return self._form
+
+        @form.setter
+        def form(self, value):
+            '''
+            :param str value: the new value for form.
+            '''
+            self._validate_form(value)
+            self._form = value
+            # Update the underlying string representation of the datatype.
+            self._parent.datatype.declaration = \
+                self._parent.write_fortran_string()
+
+        @staticmethod
+        def _validate_stencil_name(value):
+            '''Check that value is the expected stencil name.
+
+            :raises ValueError: if an invalid stencil name is found.
+
+            '''
+            const = GOceanConstants()
+            if value.lower() != const.VALID_STENCIL_NAME:
+                raise ValueError(
+                    f"The third metadata entry for a field should "
+                    f"be {const.VALID_STENCIL_NAME}(...) if it contains "
+                    f"arguments, but found '{value}'.")
+
+        @staticmethod
+        def _validate_stencil(value_list):
+            '''Check that 'value_list' is a valid list of 'stencil' elements.
+
+            :param value_list: the values to check.
+            :type value_list: List[str]
+
+            raises TypeError: if the supplied argument is not a list.
+            raises ValueError: if the supplied list is not of size 3.
+            raises TypeError: if any of the list entries are not \
+                strings.
+            raises ValueError: if any of the list entries do not \
+                conform to the expected format.
+
+            '''
+            if not isinstance(value_list, list):
+                raise TypeError(
+                    f"Stencil entries should be provided as a list, but found "
+                    f"'{type(value_list).__name__}'.")
+            if len(value_list) != 3:
+                raise ValueError(
+                    f"If the third metadata entry is a stencil, it should "
+                    f"contain 3 arguments, but found "
+                    f"{len(value_list)}.")
+            pattern = re.compile("[01]{3,3}")
+            for value in value_list:
+                if not isinstance(value, str):
+                    raise TypeError(
+                        f"Stencil entries should be strings, but found "
+                        f"'{type(value).__name__}'.")
+                if not pattern.match(value):
+                    raise ValueError(
+                        f"Stencil entries should follow the pattern "
+                        f"[01]{{3:3}}, but found '{value}'.")
+
+        @property
+        def stencil(self):
+            '''
+            :returns: the stencil value, or None if there is no stencil.
+            :rtype: Optional[List[str]]
+            '''
+            return self._stencil
+
+        @stencil.setter
+        def stencil(self, value_list):
+            '''
+            :param value_list: the new value for stencil.
```

Comment:
Updated words and added a url using markup. Not sure if that is the right thing to do, or even right format to use!

Reply:
Thanks. I think in this case it warrants having the text and link in the docstring body rather than in the :param: field. The reason I ask for this is that it makes the Reference Guide more useful.